File: android/entities/src/main/java/com/tiagohs/entities/timeline/TimelineItem.kt
package com.tiagohs.entities.timeline

import com.google.gson.annotations.SerializedName
import com.tiagohs.entities.contents.ContentInformation
import com.tiagohs.entities.image.Image
import java.io.Serializable

data class TimelineItem(
    @SerializedName("year")
    val year: String,

    @SerializedName("description")
    val description: String,

    @SerializedName("image_transparent")
    val imageTransparent: Boolean = false,

    @SerializedName("title")
    val title: String? = null,

    @SerializedName("margin_top")
    val marginTop: Int? = null,

    @SerializedName("image")
    val image: Image? = null,

    @SerializedName("image_info")
    val imageInfo: ContentInformation? = null
): Timeline(), Serializable

File: ios/HistoryCinema/HistoryCinema/Entities/LocalFiles/Timelines/TimelineItem.swift
//
//  TimelineItem.swift
//  HistoryCinema
//
//  Created by Tiago Henrique da Silva on 27/02/21.
//

import Foundation
import ObjectMapper

class TimelineItem: Timeline {
    var year: String!
    var description: String!
    var imageTransparent: Bool!
    var title: String?
    var marginTop: Int?
    var image: LocalImage?
    var imageInfo: ContentInformation?
    
    required init?(map: Map) {
        super.init(map: map)
    }

    override func mapping(map: Map) {
        super.mapping(map: map)
        
        year                                    <- map["year"]
        description                             <- map["description"]
        imageTransparent                        <- map["image_transparent"]
        title                                   <- map["title"]
        marginTop                               <- map["margin_top"]
        image                                   <- map["image"]
        imageInfo                               <- map["image_info"]
    }
}
